.github/workflows/lighthouse.yml

```diff
@@ -12,32 +12,43 @@ on:
 
 # A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
-  # This workflow contains a single job called "build"
-  build:
+  # This workflow contains a single job
+  lighthouse:
     # The type of runner that the job will run on
     runs-on: ubuntu-latest
 
     # Steps represent a sequence of tasks that will be executed as part of the job
     steps:
     # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
     - uses: actions/checkout@v2
+    - name: Setup Node
+      uses: actions/setup-node@v1
 
-    # Runs a single command using the runners shell
-    - name: Run a one-line script
-      run: echo Hello, world!
+    - name: NPM install and Build
+      run: |
+        npm i
+        npm run build
+
+    - name: Waiting for 200 from the Netlify Preview
+      uses: jakepartusch/wait-for-netlify-action@v1
+      id: waitFor200
+      with:
+        site_name: "happy-franklin-69b6d4"
 
-    # Runs a set of commands using the runners shell
-    - name: Run a multi-line script
+    # lighthouse localhost test
+    - name: Lighthouse CI
       run: |
-        echo Add other actions to build,
-        echo test, and deploy your project.
+        npm install -g @lhci/cli@0.3.x
+        lhci autorun --config=.github/lighthouse/lighthouserc.json
```

Comment:
I see that you are using lhci autorun, which takes care of starting the web server, but I still don't see the required change in package.json nor do I see the optional command line parameter to enable that automatic run.
https://github.com/GoogleChrome/lighthouse-ci/blob/master/docs/cli.md#autorun
From the docs:

Starting a webserver
To allow autorun to automatically start your webserver, add an npm script named serve:lhci.

example package.json excerpt:

  "scripts": {
    "serve:lhci": "NODE_ENV=production npm run server"
  }
Alternatively, you can supply the custom server initalization command from collect as a flag directly to autorun:

lhci autorun --collect.startServerCommand="rails server -e production"

Reply:
This PR size is now XXL. Could you please extract just the Lighthouse CI code in a separate PR so we expedite its review and merge. Thank you!

Reply:
agreed!